### PR TITLE
Skip SPI re-init if clock frequency doesn't change

### PR DESCRIPTION
**Pull request by Nerradia**
Hello,

In link with #392, I have two SPI devices that use the same clock frequency but not the same clock polarity. From what I've seen, re-initing the whole SPI is slow and unecessary when clock is the same.

I haven't done much testing but it works in my case.

### CODE REVIEW

File: libraries/SPI/src/SPI.cpp [hunk 1]
```diff
@@ -183,16 +183,20 @@ void SPIClassRP2040::beginTransaction(SPISettings settings) {
...
+    }
+    else {
+        /* Only de-init if the clock changes frequency */
+        if (settings.getClockFreq() != _spis.getClockFreq()) {
+          if (_initted) {
+              DEBUGSPI("SPI: deinitting currently active SPI\n");
+              spi_deinit(_spi);
+          }
+          DEBUGSPI("SPI: initting SPI\n");
+          spi_init(_spi, _spis.getClockFreq());
```
Comment (earlephilhower) [inline libraries/SPI/src/SPI.cpp:195]: This is backwards, actually.  `settings` is the new value (you've moved the copy `_spis = settings` after this block) so you want `spi_init(_spi, settings.getClockFreq());` I believe.

**Review by earlephilhower — CHANGES_REQUESTED** [review]

**Comment by Nerradia** [comment]
Sorry I've been wuite busy last week. I have fixed the code as you requested. I guess I can't update this PR if it is closed ?

**Comment by earlephilhower** [comment]
Try it now, please.

**Comment by Nerradia** [comment]
Alright, I think the formatting is good now.
Sorry for struggling a bit, that's my first PR !

**Review by earlephilhower — APPROVED** [review]
Thanks for getting this cleaned up!

**Pull request merged**